data/data/gcp/network/network.tf

```diff
@@ -1,58 +1,71 @@
 resource "google_compute_network" "cluster_network" {
-  name = "${var.cluster_id}-network"
+  name  = "${var.cluster_id}-network"
+  count = var.preexisting_cluster_network == "" ? 1 : 0
 
   auto_create_subnetworks = false
 }
 
+data "google_compute_network" "preexisting_cluster_network" {
+  name  = var.preexisting_cluster_network
+  count = var.preexisting_cluster_network != "" ? 1 : 0
+}
+
 resource "google_compute_subnetwork" "worker_subnet" {
   name          = "${var.cluster_id}-worker-subnet"
-  network       = google_compute_network.cluster_network.self_link
+  network       = google_compute_network.cluster_network[0].self_link
   ip_cidr_range = var.worker_subnet_cidr
+  count         = var.preexisting_cluster_network == "" ? 1 : 0
 }
 
 resource "google_compute_subnetwork" "master_subnet" {
   name          = "${var.cluster_id}-master-subnet"
-  network       = google_compute_network.cluster_network.self_link
+  network       = google_compute_network.cluster_network[0].self_link
   ip_cidr_range = var.master_subnet_cidr
+  count         = var.preexisting_cluster_network == "" ? 1 : 0
 }
 
 resource "google_compute_router" "router" {
   name    = "${var.cluster_id}-router"
-  network = google_compute_network.cluster_network.self_link
+  network = google_compute_network.cluster_network[0].self_link
+  count   = var.preexisting_cluster_network == "" ? 1 : 0
 }
 
 resource "google_compute_address" "master_nat_ip" {
-  name = "${var.cluster_id}-master-nat-ip"
+  name  = "${var.cluster_id}-master-nat-ip"
+  count = var.preexisting_cluster_network == "" ? 1 : 0
 }
 
 resource "google_compute_router_nat" "master_nat" {
   name                               = "${var.cluster_id}-nat-master"
-  router                             = google_compute_router.router.name
+  router                             = google_compute_router.router[0].name
   nat_ip_allocate_option             = "MANUAL_ONLY"
-  nat_ips                            = [google_compute_address.master_nat_ip.self_link]
+  nat_ips                            = [google_compute_address.master_nat_ip[0].self_link]
   min_ports_per_vm                   = 7168
   source_subnetwork_ip_ranges_to_nat = "LIST_OF_SUBNETWORKS"
+  count                              = var.preexisting_cluster_network == "" ? 1 : 0
 
   subnetwork {
-    name                    = google_compute_subnetwork.master_subnet.self_link
+    name                    = google_compute_subnetwork.master_subnet[0].self_link
     source_ip_ranges_to_nat = ["ALL_IP_RANGES"]
   }
 }
 
 resource "google_compute_address" "worker_nat_ip" {
-  name = "${var.cluster_id}-worker-nat-ip"
+  name  = "${var.cluster_id}-worker-nat-ip"
+  count = var.preexisting_cluster_network == "" ? 1 : 0
```

Comment:
probably replace this with a local bool `preexisting_cluster_network` rather than the empty string check on the variable.